api/v1alpha1/clientprofile_types.go

```diff
@@ -31,10 +31,15 @@ type CephFsConfigSpec struct {
 
 	//+kubebuilder:validation:Optional
 	FuseMountOptions map[string]string `json:"fuseMountOptions,omitempty"`
+
+	//+kubebuilder:validation:XValidation:rule="self == oldSelf",message="radosNamespace is immutable"
+	//+kubebuilder:validation:Optional
+	RadosNamespace string `json:"radosNamespace,omitempty"`
```

Comment:
When it comes to rados namespaces, the empty string is a valid namespace name and is used for the implicit namespace. In this case an optional value should be backed by a pointer to a string to differentiate the implicit namespace from non setting a namespace at all that will default to the behavior which might or might not be using the implicit namespace

Reply:
cephcsi doesnt support empty namespace, am fine with having it as a pointer